fs/time.go

```diff
@@ -0,0 +1,21 @@
+package fs
+
+import (
+	"syscall"
+	"time"
+)
+
+// Gnu tar and the go tar writer don't have sub-second mtime
+// precision, which is problematic when we apply changes via tar
+// files, we handle this by comparing for exact times, *or* same
+// second count and either a or b having exactly 0 nanoseconds
+func sameFsTime(a, b time.Time) bool {
+	return a == b ||
+		(a.Unix() == b.Unix() &&
+			(a.Nanosecond() == 0 || b.Nanosecond() == 0))
+}
+
+func sameFsTimeSpec(a, b syscall.Timespec) bool {
```

Comment:
Yes, the original ported change code used it. I think this is the incorrect logic which causes https://github.com/docker/docker/issues/21555 so let's not keep it around :)